http-server/src/Main.gren

```diff
@@ -0,0 +1,94 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Node.Program as Program exposing (Program)
+import Node.Http as Http
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { stdout : Stream
+    , stderr : Stream
+    }
+
+
+type Msg
+    = CreateServerResult (Result Http.CreateServerError Http.Server)
+    | HandleRequest Http.Request
+    {-
+        type alias Request =
+            { headers : Dict String String
+            , body : Bytes
+            , url : Url
+            , method : Method
+            }
+        
+        type Method
+            = GET
+            | HEAD
+            | POST
+            | PUT
+            | DELETE
+            | CONNECT
+            | TRACE
+            | PATCH
+    -}
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Node.initialize <| \nodeConfig ->
+        Program.await Http.initialize <| \serverPermission ->
+            Program.startProgram
+                { model =
+                    { stdout = nodeConfig.stdout
+                    , stderr = nodeConfig.stderr
+                    }
+                , command =
+                    Http.createServer serverPermission
+                        { host = "localhost"
+                        , port = "8765"
+                        }
+                        |> Task.attempt CreateServerResult
+                }
+
+
+update : Msg -> Model -> { model : Model, command : Cmd Msg }
+update msg model =
+    case msg of
+        CreateServerResult result ->
+            case result of
+                Ok _ ->
+                    { model = model
+                    , command = Stream.sendString model.stdout
+                        "Server started"
+                    }
+                Err _ ->
+                    { model = model
+                    , command = Stream.sendString model.stderr
+                        "Server failed to start"
+                    }
+        HandleRequest req ->
+            { model = model
+            , command = Http.respondWith req
+                { status = 200
+                , body = BE.encode (BE.string "<html>Hello</html>")
+                , headers = Dict.fromArray
+                    [ { key = "Content-type", value = "text/html" }
+                    ]
+                }
+            }
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Http.onRequest HandleRequest
```

Comment:
...maybe this should accept the server from the result of `Http.createServer`?

Reply:
I guess it would need to, since it's possible to start multiple servers. It also prevents the weird situation of subscribing to a server that doesn't exist.

Reply:
updated to reflect this